sklearn/metrics/_classification.py

```diff
@@ -2520,6 +2527,16 @@ def balanced_accuracy_score(y_true, y_pred, *, sample_weight=None, adjusted=Fals
         performance would score 0, while keeping perfect performance at a score
         of 1.
 
+    zero_division : {"warn", 0.0, 1.0, np.nan}, default="warn"
+        Sets the value to return when there is a zero division.
+
+        Notes:
+
+        - If set to "warn", this acts like 0, but a warning is also raised.
+        - If set to `np.nan`, such values will be excluded from the average.
+
+        .. versionadded:: 1.6
```

Comment:
I think the docstring as is, is vague and doesn't give enough information.

For instance, this warning is now removed: `y_pred contains classes not in y_true`. 

The diff here seems more complicated than the other instances of adding `zero_division`, and this parameter is doing more than simply replacing a nan of a division by zero.

Better explaining the actual behavior, it'd be more clear how to proceed.

Reply:
We could improve the current warning to give more info - both warings (old and new) are correct: when `y_pred` contains classes not in `y_true`, tp + fn = 0 (sum of true labels) for that class, this results in zero division and balance accuracy is ill-defined in this case.

How about we improve the warning to give more info, e.g., it is ill-defined because y_pred contains classes not in y_true (resulting in zero division when calculating recall)?

I appreciate it is not similar to precision/recall/f1 where there is explicit averaging/an explicit `average` parameter, so it's not so obvious that averaging is happening. We do say above: "It is defined as the average of recall obtained on each class" but we could be more explicitly explain in `zero_division` parameter that the zero division occurs when calculating recall for each class?

Similar to what we do in `cohen_kappa_score`:

https://github.com/scikit-learn/scikit-learn/blob/56bbb5aeda441f3d5c129cd56c65bb1f09d7bb65/sklearn/metrics/_classification.py#L740-L744

(though `cohen_kappa_score` is even more different and there is no averaging)